components/authz-service/storage/v2/rule.go

```diff
@@ -72,8 +72,10 @@ func validateRuleInputs(id string,
 	}
 
 	for _, condition := range conditions {
-		if condition.Type != ruleType {
-			return errors.New("all condition types must match the parent rule type")
+		if ruleType == Event {
+			if condition.Attribute != Organization && condition.Attribute != ChefServer {
+				return errors.New("rules of type Event only accept Conditions with attributes Organization or ChefServer")
+			}
```

Comment:
While this gets the job done, it is what I think of as "magic expressions" (akin to magic strings). Would be a bit cleaner to do something like this:
```
if err := conditionAllowed(ruleType, condition.Attribute); err != nil {
	return err
}
```
and `conditionAllowed` could be a tad more explicit with something like:
```
if ruleType == Event {
  if condition.Attribute != Organization && condition.Attribute != ChefServer {
	return errors.New("rules of type Event only accept Conditions with attributes Organization or ChefServer")
  }
} else if ruleType == Node {
  return nil // all values allowed for nodes
}
```